File: ios/QRScanManager.m
//
//  QRScanManager.m
//  RNQRCode
//
//  Created by HungDV on 7/2/20.
//

#if __has_include(<React/RCTViewManager.h>)
#import <React/RCTViewManager.h>
#elif __has_include(“RCTViewManager.h”)
#import “RCTViewManager.h”
#else
#import “React/RCTViewManager.h” // Required when used as a Pod in a Swift project
#endif


#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(QRScan, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(enableOverlay, BOOL)
RCT_EXPORT_VIEW_PROPERTY(enableDelay, BOOL)
RCT_EXPORT_VIEW_PROPERTY(delayReceive, NSInteger)
RCT_EXPORT_VIEW_PROPERTY(onQRCodeRead, RCTDirectEventBlock)

RCT_EXTERN_METHOD(stopPreview: (nonnull NSNumber *) node)
RCT_EXTERN_METHOD(resumePreview: (nonnull NSNumber *) node)
@end

File: ios/QRScanManager.swift
//
//  QRScanManager.swift
//  RNQRCode
//
//  Created by HungDV on 7/2/20.
//

import Foundation
import UIKit
//import QRScanView

@objc(QRScan)
class QRScanManager : RCTViewManager {
  // Return the native view that represents your React component
  override func view() -> UIView! {
    return QRScanView()
  }

  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
//  
  @objc func stopPreview(_ node: NSNumber) {
    DispatchQueue.main.async {
      let view = self.bridge.uiManager.view(forReactTag: node) as! QRScanView
      view.stopScanning()
    }
  }
  
  @objc func resumePreview(_ node: NSNumber) {
    DispatchQueue.main.async {
      let view = self.bridge.uiManager.view(forReactTag: node) as! QRScanView
      view.startScanning()
    }
  }

}
